xarray/core/indexing.py

```diff
@@ -249,6 +249,8 @@ def remap_label_indexers(data_obj, indexers, method=None, tolerance=None):
                                  'an associated coordinate.')
             pos_indexers[dim] = label
         else:
+            coords_dtype = data_obj.coords[dim].values.dtype
```

Comment:
Could we simplify this as just:
```suggestion
            coords_dtype = data_obj.coords[dim].dtype
```